pkg/registry/apis/provisioning/webhook.go

```diff
@@ -95,17 +95,27 @@ func (s *webhookConnector) Connect(ctx context.Context, name string, opts runtim
 			return
 		}
 		if rsp.Job != nil {
+			if rsp.JobName == "" {
+				switch rsp.Job.Action {
+				case provisioning.JobActionSync, provisioning.JobActionMigrate:
+					// Sync and migrate jobs should never run at the same time. Hence, the name encapsulates them both (and the spec differentiates them).
+					rsp.JobName = fmt.Sprintf("%s-syncmigrate", repo.Config().GetName())
+				default:
+					rsp.JobName = fmt.Sprintf("%s-%s", repo.Config().GetName(), rsp.Job.Action)
```

Comment:
umm. interesting. See how you dealt with the naming here as I mentioned in `github.go`. To many places for building the name.